src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs

```diff
@@ -1069,17 +1069,19 @@ private bool HasHeader
         /// <param name="allowSparse">Whether the file can contain numerical vectors in sparse format.</param>
         /// <param name="allowQuoting">Whether the content of a column can be parsed from a string starting and ending with quote.</param>
         /// <param name="dataSample">Allows to expose items that can be used for reading.</param>
+        /// <param name="trimWhitespace">Remove trailing whitespace from lines.</param>
         internal TextLoader(IHostEnvironment env, Column[] columns, char separatorChar = Defaults.Separator,
             bool hasHeader = Defaults.HasHeader, bool allowSparse = Defaults.AllowSparse,
-            bool allowQuoting = Defaults.AllowQuoting, IMultiStreamSource dataSample = null)
-            : this(env, MakeArgs(columns, hasHeader, new[] { separatorChar }, allowSparse, allowQuoting), dataSample)
+            bool allowQuoting = Defaults.AllowQuoting, IMultiStreamSource dataSample = null, bool trimWhitespace = Defaults.TrimWhitespace)
```

Comment:
>IMultiStreamSource dataSample = null [](start = 55, length = 36)

If you wanted you could just get rid of this constructor opportunistically. It serves no purpose as far as I can tell.

Reply:
Dropped.

---
In reply to: [259963695](https://github.com/dotnet/machinelearning/pull/2710#discussion_r259963695) [](ancestors = 259963695)